sensors/sensors.go

```diff
@@ -67,31 +91,99 @@ func NewLidar(
 	}, nil
 }
 
-// ValidateGetData checks every sensorValidationIntervalSec if the provided lidar
-// returned a valid timed lidar readings every sensorValidationIntervalSec
+// NewIMU returns a new IMU.
+func NewIMU(
+	ctx context.Context,
+	deps resource.Dependencies,
+	imuName string,
+	logger golog.Logger,
+) (IMU, error) {
+	_, span := trace.StartSpan(ctx, "viamcartographer::sensors::NewIMU")
+	defer span.End()
+	if imuName == "" {
+		logger.Info("no movement sensor configured, proceeding without IMU")
+		return IMU{}, nil
+	}
+	newIMU, err := movementsensor.FromDependencies(deps, imuName)
+	if err != nil {
+		return IMU{}, errors.Wrapf(err, "error getting IMU movement sensor %v for slam service", imuName)
+	}
+
+	// A movement_sensor used as an IMU must support LinearAcceleration and Angular Velocity.
+	properties, err := newIMU.Properties(ctx, make(map[string]interface{}))
+	if err != nil {
+		return IMU{}, errors.Wrapf(err, "error getting movement sensor properties %v for slam service", imuName)
+	}
+	if !(properties.LinearAccelerationSupported && properties.AngularVelocitySupported) {
+		return IMU{}, errors.New("configuring IMU movement sensor error: " +
```

Comment:
[nit/opt] A better way to do multiline strings is using backticks. Example:
```
`hi
second line`
```

Reply:
None of the existing code does this... want me to change it everywhere?

Reply:
hmmm, fair point. Lets keep it as is and have a separate PR for that.

Reply:
Want me to make the PR?

Reply:
Spoke offline, future refactor

Reply:
I will make a JIRA ticket for this